### DISCUSSION TITLE
Unable to Display Dates from Database in WorkDatesDisplay component

### DISCUSSION THREAD

**Discussion by migueljcpinto** [neuefische/questions#708] [Web] [answered]
Hi there!! 

On my project I have a WorkDatesDisplay component that is supposed to display a list of dates fetched from a MongoDB database. The data structure from the database consists of absence dates, each with a type (either "vacation" or "dayOff") and a date field. 

I've fetched the absence records and passed them as a prop (absenceDates) to the WorkDatesDisplay component.
Inside the component, I've filtered the absenceDates based on the type to create two separate lists: vacationDates and daysOff.
I've attempted to map over these lists to display the dates - but without success.

The dates are not being displayed in the component, even though I can see the data in console logs. I'm also getting duplicates all the time, i.e. the code is rendered twice.

I'd be grateful for any help.

[PR](https://github.com/MiketheBite/Capstone-project/pull/15)

**Comment by Kheirah** [comment, marked as answer]
Currently, you have `date` as the type of your date field. In the `Absence` schema, try changing the type of your date field to an array of Dates.
```js
const absenceSchema = new Schema({
  nurseId: { type: Schema.Types.ObjectId, ref: "Nurse", required: true },
  type: { type: String, enum: ["dayOff", "vacation"], required: true },
  date: { type: [Date], required: true }, //this line is changed
});
```
This could be the reason why you can't map over that field since it's not an array.

**Reply by migueljcpinto** [reply]
Right, and I was also passing a wrong prop. `absenceDates={absencesData}`
Thanks Master :D
